service/s3/api_client.go

```diff
@@ -359,7 +371,7 @@ func (c *PresignClient) convertToPresignMiddleware(stack *middleware.Stack, opti
 	}
 	// add middleware to set expiration for s3 presigned url, if expiration is set to
 	// 0, this middleware sets a default expiration of 900 seconds
-	err = stack.Build.Add(&s3cust.AddExpiresOnPresignedURL{Expires: c.expires}, middleware.After)
+	err = stack.Build.Add(&s3cust.AddExpiresOnPresignedURL{Expires: c.Expires}, middleware.After)
 	if err != nil {
 		return err
 	}
```

Comment:
learnings from waiter implementation: since time.Duration is alias for int64.

Here c.Expires can be negative. The `AddExpiresOnPresignedURL` middleware does not handle the negative expiration case.  Should return an error.